File: seed/csharp-sdk/exhaustive/no-generate-error-types/src/SeedExhaustive/ReqWithHeaders/Requests/ReqWithHeaders.cs
using System.Text.Json.Serialization;
using SeedExhaustive.Core;

namespace SeedExhaustive;

[Serializable]
public record ReqWithHeaders
{
    [JsonIgnore]
    public required string XTestServiceHeader { get; set; }

    [JsonIgnore]
    public required string XTestEndpointHeader { get; set; }

    [JsonIgnore]
    public required string Body { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/php-sdk/exhaustive/wire-tests/src/ReqWithHeaders/Requests/ReqWithHeaders.php
<?php

namespace Seed\ReqWithHeaders\Requests;

use Seed\Core\Json\JsonSerializableType;

class ReqWithHeaders extends JsonSerializableType
{
    /**
     * @var string $xTestServiceHeader
     */
    public string $xTestServiceHeader;

    /**
     * @var string $xTestEndpointHeader
     */
    public string $xTestEndpointHeader;

    /**
     * @var string $body
     */
    public string $body;

    /**
     * @param array{
     *   xTestServiceHeader: string,
     *   xTestEndpointHeader: string,
     *   body: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->xTestServiceHeader = $values['xTestServiceHeader'];
        $this->xTestEndpointHeader = $values['xTestEndpointHeader'];
        $this->body = $values['body'];
    }
}
